backend/de.metas.shipper.gateway.dhl/src/main/java/de/metas/shipper/gateway/dhl/DhlShipperGatewayClient.java

```diff
@@ -273,250 +167,238 @@ private Object doActualRequest(final Object request, @NonNull final DeliveryOrde
-				//						docPosition.setCustomsValue(customsDocument.getCustomsValue());
-				//						exportDocumentType.getExportDocPosition().add(docPosition);
-				//
-				//						shipmentOrderTypeShipment.setExportDocument(exportDocumentType);
-				//					}
-				//				}
-
-				// (2) create the needed shipment order type
-				final ShipmentOrderType shipmentOrderType = objectFactory.createShipmentOrderType();
-				final DhlCustomDeliveryData dhlCustomDeliveryData = DhlCustomDeliveryData.cast(deliveryOrder.getCustomDeliveryData());
-				final DhlCustomDeliveryDataDetail deliveryDetail = dhlCustomDeliveryData.getDetailByPackageId(packageIdsAsList.get(i).getRepoId());
-				shipmentOrderType.setSequenceNumber(deliveryDetail.getSequenceNumber().getSequenceNumber());
-				shipmentOrderType.setShipment(shipmentOrderTypeShipment);
-				createShipmentOrderRequest.getShipmentOrder().add(shipmentOrderType);
+				throw createShipperException(dhlOrderRequest, result.getStatusCode(), String.valueOf(result.getBody()));
 			}
+			return result.getBody();
 		}
+		catch (final HttpClientErrorException e)
+		{
+			final HttpStatus statusCode = e.getStatusCode();
+			final String responseBodyAsString = e.getResponseBodyAsString();
+			throw createShipperException(dhlOrderRequest, statusCode, responseBodyAsString);
+		}
+	}
 
-		return createShipmentOrderRequest;
+	private AdempiereException createShipperException(final @NonNull JSONDhlCreateOrderRequest dhlOrderRequest, @NonNull final HttpStatus statusCode, @NonNull final String responseBodyAsString)
+	{
+		final ShipperGatewayException shipperGatewayException = //
+				new ShipperGatewayException("HttpClientErrorException with statusCode=" + statusCode);
+
+		return AdempiereException.wrapIfNeeded(shipperGatewayException)
+				.appendParametersToMessage()
+				.setParameter("responseBodyAsString", responseBodyAsString)
+				.setParameter("routingRequest", dhlOrderRequest);
 	}
 
-	@NonNull
-	private WebServiceTemplate createWebServiceTemplate()
+	private JSONDhlCreateOrderRequest createNewDHLShipmentOrderRequest(@NonNull final DeliveryOrder deliveryOrder)
 	{
-		// cig authentication (basic http authentication) required @ each request: https://entwickler.dhl.de/en/group/ep/authentifizierung
-		final UsernamePasswordCredentials credentials = new UsernamePasswordCredentials(config.getApplicationID(), config.getApplicationToken());
+		final JSONDhlCreateOrderRequest.JSONDhlCreateOrderRequestBuilder orderRequestBuilder = JSONDhlCreateOrderRequest.builder()
+				//I can't tell if this is variable in any way, but on the sandbox environment this is the only profile I was able to use
+				.profile(DhlConstants.STANDARD_GRUPPENPROFIL);
+		final ImmutableList.Builder<JsonDhlShipment> shipments = ImmutableList.builder();
+		final ContactPerson deliveryContact = deliveryOrder.getDeliveryContact();
+		for (final DeliveryOrderLine deliveryOrderLine : deliveryOrder.getDeliveryOrderLines()) // only a single delivery position should exist
```

Comment:
I would use "order line" instead of "position" for the sake of a ubiquitous language

Reply:
removed comment altogether.